index.js

```diff
@@ -29,20 +33,104 @@ cp.parseBlock = function () {
 	return this.finishNode(node, "BlockStatement");
 };
 
+function strToModuleIdentifier(str) {
+	return '$m_' + str.replace(/-(\w)/g, (_, letter) => letter.toUpperCase()).replace(/[^$_a-zA-Z0-9]/g, '_');
+}
+
+function writePrefix (imports) {
+	let transformed = `Object.defineProperty(exports, '__esModule', { value: true });\n`;
```

Comment:
@jdalton Do we want to follow Babel's path here even for minimalistic in-place transpiler? I think we can assume that our module will be imported only by same our transpiler, and that our transpiler is importing everything in ES6 style - this wy we will not need to be bound to issues around CommonJS <-> ESM interop.